### PR TITLE
Add support of SPI API for STM32H5 family

### PR DESCRIPTION
**Pull request by JohnK1987**
### Summary of changes 
This PR will add support of SPI API for STM32H5 family

### Documentation 

----------------------------------------------------------------------------------------------------------------
### Pull request type 

    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results 
I performed just test with SD card read/write.

    [] No Tests required for this change (E.g docs only update)
    [] Covered by existing mbed-os tests (Greentea or Unittest)
    [x] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: targets/TARGET_STM/TARGET_STM32H5/spi_device.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#ifndef MBED_SPI_DEVICE_H
+#define MBED_SPI_DEVICE_H
+
+#include "stm32h5xx_ll_rcc.h"
+#include "stm32h5xx_ll_spi.h"
+
+#define SPI_IP_VERSION_V2
+
+// Defines the word legnth capability of the device where Nth bit allows for N window size
+#define STM32_SPI_CAPABILITY_WORD_LENGTH (0xFFFFFFF8)
```
Comment (JohnK1987) [inline targets/TARGET_STM/TARGET_STM32H5/spi_device.h:25]: Not sure about this value

File: targets/TARGET_STM/stm_spi_api.c [hunk 2]
```diff
@@ -314,8 +316,10 @@ static void _spi_init_direct(spi_t *obj, const spi_pinmap_t *pinmap)
         PeriphClkInit.PeriphClockSelection = RCC_PERIPHCLK_SPI2;
 #if defined (RCC_SPI123CLKSOURCE_PLL)
         PeriphClkInit.Spi123ClockSelection = RCC_SPI123CLKSOURCE_PLL;
-#else
+#elif defined (RCC_SPI2CLKSOURCE_SYSCLK)
```
Comment (JohnK1987) [inline targets/TARGET_STM/stm_spi_api.c:319]: I have to test if this works properly for rest targets

File: targets/TARGET_STM/TARGET_STM32H5/spi_device.h (hunk 1 above)
Comment (multiplemonomials) [reply targets/TARGET_STM/TARGET_STM32H5/spi_device.h:25]: Well that's pretty annoying... only some of the SPIs have 32 bit support 
image
Probably at some point we'll need to make the code a little smarter here but for now this value should be OK

**Comment by JohnK1987** [comment]
> * spi-microsd works for the serial implementation but infinite hangs for interrupt and DMA transfers
> * spi-slave-comms passes except for the test cases which use SPI modes 1 and 2

Ok, and is this related just to STM32H5 or complete STM32?

**Comment by multiplemonomials** [comment]
I'm going to go ahead and merge this for now -- though it still is exhibiting some issues with the DMA SPI stuff, I think that that is more to do with the DMA code and not with this specific platform port specifically.

**Review by multiplemonomials — APPROVED** [review]

**Pull request merged**